frontend/src/components/messages/ReplyRequestDialog.vue

```diff
@@ -97,28 +98,68 @@ export default {
     },
 
     /** 
-     * Create the reply and emit success event.
+     * Create the reply, update status to assigned, and emit success event.
      */
     async submit() {
-      this.$refs.replyRequestForm?.validate()
-      if (this.isFormComplete) {
+      if (this.isFormComplete && this.$refs.replyRequestForm?.validate()) {
         try {
           this.isLoading = true
-          const payload = {
-            assistanceRequestId: this.assistanceRequestId,
-            message: this.message,
-          }
-          await this.replyToAssistanceRequest(payload)
+          this.createReply(this.assistanceRequestId)
+          this.updateStatusToAssigned(this.assistanceRequestId)
+          this.updateStoredAssistanceRequest(this.assistanceRequestId)
```

Comment:
You need to await each of these if you want them to execute in order.

Reply:
Fixed.